int32 number
string color
